### ISSUE TITLE
Is it possible to use whisper with metamask?

### ISSUE THREAD

**Issue by yaronvel** [MetaMask/metamask-plugin#895]
If all users are connected to a centeral node, is it possible to user whisper, even though nodes to not relay whisper msgs?

**Comment by flyswatter** [comment]
Much like Swarm, Whisper has the drawback for MetaMask that it does not use a browser-compatible networking standard.

To integrate the Whisper API, we would need a sort of bridged client running on clients that can talk to both normal Whisper users as well as a browser-compatible network (like WebRTC).

This is the same architecture pattern that WebTorrent achieved for BitTorrent, and I really wish people would start developing protocols with browsers in mind in the first place ;)

This is a great goal long term, but there are a few layers of infrastructure that we need built before we can add this easily to MetaMask.

I wonder if @diasdavid has thought about this yet.

**Comment by kumavis** [comment]
We are working on a libp2p based p2p network that will enable light client and alternate protocols. however, the road to this working is a long one.

**Comment by kumavis** [comment]
TL;DR: cant use whisper with metamask. it may work some day.

**Comment by flyswatter** [comment]
If it's a "someday" issue, this might belong in the `Future` milestone, instead of `closed`?

**Comment by thorstenhirsch** [comment]
So 6 years later... any updates on this issue?

**Comment by danfinlay** [comment]
The short answer is today whisper cannot run on a browser alone, because it uses tcp/udp. If someone made a bridge protocol, or if we eventually had say a desktop copilot app, those could provide access to the whisper network, but currently neither has been done by us, but we have been looking at the desktop platform as a possibility for some capabilities like that.
